packages/loader/container-loader/src/serializedStateManager.ts

```diff
@@ -198,7 +216,17 @@ export class SerializedStateManager {
 		}
 	}
 
-	public async refreshLatestSnapshot(supportGetSnapshotApi: boolean) {
+	/**
+	 * Fetch the latest snapshot for the container, including delay-loaded groupIds if pendingLocalState was provided and contained any groupIds.
+	 *
+	 * If there are no groupIds, then we will attempt to update the class's snapshot and processedOps.
```

Comment:
This seems wrong.  I would assume we should always call `updateSnapshotAndProcessedOpsMaybe`.  I'm not sure how the delay-loaded data fits in though.  It's returned, but no one calls this directly to do anything with the return value.

Reply:
> I'm not sure how the delay-loaded data fits in though

Like, should it be saved in a class variable alongside `this.snapshot`?

Reply:
When we call closeAndGetPendingOps (you can trace from there), that's when we eventually grab the locally stored data.

Reply:
This does call `updateSnapshotAndProcessedOpsMaybe` at the end of the call here.

Reply:
> This does call updateSnapshotAndProcessedOpsMaybe at the end of the call here.

As discussed offline, there was an early return.  I fixed it here.

> When we call closeAndGetPendingOps (you can trace from there), that's when we eventually grab the locally stored data.

Ohhhhh the StorageAdapter is caching it. Did not understand or expect that.  So the return value can just be dropped altogether.